core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/StrictEvaluationStrategy.java

```diff
@@ -1823,15 +1824,158 @@ public Value evaluate(ValueExprTripleRef node, BindingSet bindings)
 
 	/**
 	 * evaluates a TripleRef node returning bindingsets from the matched Triple nodes in the dataset (or explore
-	 * standart reification)
+	 * standard reification)
 	 * 
 	 * @param ref      to evaluate
 	 * @param bindings with the solutions
 	 * @return iteration over the solutions
 	 */
 	public CloseableIteration<BindingSet, QueryEvaluationException> evaluate(TripleRef ref, BindingSet bindings) {
-		// not supported
-		return new EmptyIteration<BindingSet, QueryEvaluationException>();
+		// Naive implementation that walks over all statements matching (x rdf:type rdf:Statement)
+		// and filter those that do not match the bindings for subject, predicate and object vars (if bound)
+		final org.eclipse.rdf4j.query.algebra.Var subjVar = ref.getSubjectVar();
+		final org.eclipse.rdf4j.query.algebra.Var predVar = ref.getPredicateVar();
+		final org.eclipse.rdf4j.query.algebra.Var objVar = ref.getObjectVar();
+		final org.eclipse.rdf4j.query.algebra.Var extVar = ref.getExprVar();
```

Comment:
I'm not sure I follow where this `extVar` comes from. Is this intended to encapsulate use of RDF* triples in things like BIND clauses? Why not use a BindingSetAssignment for that case?

Reply:
Not sure how BindingSetAssignment can be re-used. 

Each reference to a Triple in the SPARQL* syntax is translated (recursively) to a TripleRef that provides bindings (instances of Triple) through it's `extVar` and that ext var is used at the place where the reference appear. E.g.:
 - the Statement pattern `<< ?s ?p ?o>> ?predicate ?object .` is translated to a Join between [TripleRef(?s ?p?o) to ?anon_ext] and [?anon_ext ?predicate ?object]
or in the case of bind:
 - `BIND(<<?s ?p ?o>> as ?t)` is translated to [TripleRef(?s ?p ?o) to ?anon_ext] and [BIND(?anon_ext as ?t)]

Reply:
Of course, that makes sense, thanks for clarifying!